contrib/context_interceptor/src/main/java/io/opentelemetry/contrib/context/interceptor/Log4JMdcContextStorageListener.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.context.interceptor;
+
+import io.grpc.Context;
+import io.grpc.override.ContextStorageListener;
+import io.opentelemetry.trace.SpanContext;
+import io.opentelemetry.trace.unsafe.ContextUtils;
+import org.apache.log4j.MDC;
+
+public class Log4JMdcContextStorageListener implements ContextStorageListener.Provider {
+  /**
+   * Context key for the current trace ID. The name is {@value}.
+   *
+   * @since 0.17
+   */
+  public static final String TRACE_ID_CONTEXT_KEY = "traceId";
+
+  /**
+   * Context key for the current span ID. The name is {@value}.
+   *
+   * @since 0.17
+   */
+  public static final String SPAN_ID_CONTEXT_KEY = "spanId";
+
+  @Override
+  public ContextStorageListener create() {
+    return new ContextStorageListener() {
+      @Override
+      public void contextUpdated(Context oldContext, Context newContext) {
+        SpanContext current = ContextUtils.getValue(newContext).getContext();
+        if (current.isValid()) {
+          MDC.put(TRACE_ID_CONTEXT_KEY, current.getTraceId().toLowerBase16());
+          MDC.put(SPAN_ID_CONTEXT_KEY, current.getSpanId().toLowerBase16());
+        } else {
+          MDC.put(TRACE_ID_CONTEXT_KEY, null);
```

Comment:
Sorry about rising this question again, but i did some investigation and found that 
https://github.com/innoq/log4j/blob/master/src/main/java/org/apache/log4j/MDC.java#L141
using http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/Hashtable.java
and when we put null there 
```
    public synchronized V put(K key, V value) {

        // Make sure the value is not null

        if (value == null) {

            throw new NullPointerException();

        }
```
also i didn't find tests for this branch of if

Reply:
this is still a prototype, but I changed it. I think it was tested, wasn't it?

Reply:
Yeah i know that it just a prototype, but after people start to do own interceptors this will save some time for them

Reply:
Also for tests i was looking at Log4JMdcContextStorageListenerTest class maybe i missed something